packages/mdc-list/index.js

```diff
@@ -80,22 +85,55 @@ export class MDCList extends MDCComponent {
 
   /** @return Array<!Element>*/
   get listElements_() {
-    return [].slice.call(this.root_.querySelectorAll(strings.ITEMS_SELECTOR))
-      .filter((ele) => ele.parentElement === this.root_);
+    return [].slice.call(this.root_.querySelectorAll(strings.ITEMS_SELECTOR));
   }
 
   /** @param {boolean} value */
   set wrapFocus(value) {
     this.foundation_.setWrapFocus(value);
   }
 
+  /** @param {boolean} isSingleSelectionList */
+  set singleSelection(isSingleSelectionList) {
+    if (isSingleSelectionList) {
+      this.root_.addEventListener('click', this.handleClick_);
```

Comment:
This should be a foundation method.

Reply:
? Aren't we moving toward adding event listeners in the index.js ? See `initialSyncWithDOM` above.

Reply:
True, we're exploring it. I guess we can try it here and see how it goes.

Will `handleClick` cause problems if it's triggered by a non-single selection list item? If so, there should be a check inside the `handleClick` method that ensures it won't cause problems if it's called.

Reply:
`handleClick` has logic to determine if you did not click on a list item (it looks for an ancestor `mdc-list-item`) and if doesn't find one it returns early.